applications/StructuralMechanicsApplication/custom_elements/small_displacement_shifted_boundary_element.h

```diff
@@ -0,0 +1,314 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:         BSD License
+//                   license: StructuralMechanicsApplication/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//
+
+#pragma once
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "includes/element.h"
+
+// Application includes
+#include "small_displacement.h"
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+template<std::size_t TDim>
```

Comment:
No doxygen. Why Template in element but not in condition?, why only dimension and not the number of nodes?

Reply:
The condition uses the base Kratos `Geometry`, for which the dimension and number of nodes make no sense. Indeed, the number of nodes is completely dependent on the cloud of points used for the calculation of the extension operator.

About the Doxygen, I'm adding it right away.